parsl/executors/high_throughput/interchange.py

```diff
@@ -501,6 +519,8 @@ def start(self, poll_period=None):
 
                     self.results_outgoing.send_multipart(b_messages)
                     logger.debug("[MAIN] Current tasks: {}".format(self._ready_manager_queue[manager]['tasks']))
+                    if len(self._ready_manager_queue[manager]['tasks']) == 0:
```

Comment:
Ok, this logic makes sense. The only point where a manager can go idle is when a result is returned and #tasks == 0